exercises/queen-attack/canonical-data.json

```diff
@@ -240,6 +240,26 @@
             }
           },
           "expected": true
+        },
+        {
+          "uuid": "543f8fd4-2597-4aad-8d77-cbdab63619f8",
+          "description": "cannot attack with equal difference",
```

Comment:
Not sure I understand what is expected by the test from this description. Is it possible to rephrase it in chess board terms?

Reply:
Some suggestions, ordered by length

* cannot attack if falling diagonals are only the same when reflected across the longest falling diagonal
* cannot attack if falling diagonals are on the two opposite sides of the longest falling diagonal
* cannot attack on opposite falling diagonals
    * Shorter isn't necessarily better. It is not immediately obvious what an "opposite" diagonal is, if it's not made clear that it's in relation to the longest falling diagonal.

Reply:
> cannot attack if falling diagonals are on the two opposite sides of the longest falling diagonal

FWIW: ^ this description is most understandable to me, and definitely a lot more descriptive than my initial one.

Reply:
I'll expand a bit on the situation under test, to hopefully make it clearer.

You'll note the two positions (4,1) and (2,5) in the new test are on opposite sides of the longest falling diagonal – the one from the top left corner to the bottom right corner of the board.

Reflecting position (4,1) to the other side of this longest falling diagonal would make it (1,4), which is diagonally next to (2,5).

Similarly, reflecting (2,5) to the other side of this longest falling diagonal would make it (5,2), which is diagonally next to (4,1).

A buggy implementation, such as the one I listed, would falsely claim that (4,1) and (2,5) are in a diagonal, because the reflection of either point _is_ diagonal of the respective other point.

This new test makes sure that an implementation with this particular bug does not make it through.

Does this help?

Also, since I used "reflect" a lot, allow me to change my mind and now think the first suggestion is the better one:
> cannot attack if falling diagonals are only the same when reflected across the longest falling diagonal